src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/ParameterExpander.java

```diff
@@ -704,13 +706,14 @@ public String getBuildStartedMessage(Run build, TaskListener listener, ChangeBas
      * @return the message
      */
     protected String findMessage(String completedCommand) {
-        String messageStart = "--message '";
-        String fromMessage = completedCommand.substring(completedCommand.indexOf(messageStart));
-        int endIndex = fromMessage.indexOf("' --");
-        if (endIndex <= -1) {
-            endIndex = fromMessage.length();
+        String message = "";
+        String messageRegex = "(?s)--message\\s+'(.*?)'";
+        Pattern p = Pattern.compile(messageRegex);
+        Matcher m = p.matcher(completedCommand);
+        while (m.find()) {
```

Comment:
Sorry but why use `while` instead of `if`?

Reply:
A message can be defined more then one time in commands, by a mistake.
In this situation we have at least three cases of how to act:

1. to choose the leftmost message, using `if`;
2. to choose the rightmost message, using `while`;
3. to compose a message out of all the definitions.

I supposed the second case could be the best one here.